backend/tests/test_endpoints.py

```diff
@@ -0,0 +1,101 @@
+from fastapi.testclient import TestClient
+import pytest
+from rope.api.main import app
+from rope.api.sessions import get_request_session, sessions
+from rope.api.database import SessionLocal
+from rope.db.schema import UserAccount
+
+
+@pytest.fixture
+def test_client():
+    return TestClient(app)
+
+
+@pytest.fixture
+def db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@pytest.fixture(autouse=True)
+def clear_database_table(db):
+    db.query(UserAccount).delete()
+    db.commit()
+
+
+def override_get_request_session():
+    session_id = {"session_id": "12345"}
+    return session_id
+
+
+def test_get_current_user(test_client, mocker):
+    app.dependency_overrides[get_request_session] = override_get_request_session
+    user = {
+        "12345": {
+            "email": "test@rice.edu",
+            "is_manager": False,
+            "is_admin": False,
+        }
+    }
+    mocker.patch(
+        "rope.api.sessions.sessions",
+        user,
+    )
+    response = test_client.get("/user/current")
+    data = response.json()
+    assert response.status_code == 200
+    assert data["email"] == "test@rice.edu"
+    assert data["is_manager"] is False
+    assert data["is_admin"] is False
+
+
+def test_unauthenticated_user(test_client, mocker):
+    app.dependency_overrides[get_request_session] = override_get_request_session
+    user = {
+        "ABCDEFG": {
+            "email": "test@rice.edu",
+            "is_manager": False,
+            "is_admin": False,
+        }
+    }
+    mocker.patch(
+        "rope.api.sessions.sessions",
+        user,
+    )
+    response = test_client.get("/user/current")
+    assert response.status_code == 401
+
+
+def test_delete_session(test_client, mocker):
+    app.dependency_overrides[get_request_session] = override_get_request_session
+    mocker.patch(
```

Comment:
Minor: You can make a small change to this test to also assert the expected change is made to the session store:

```python
    session_store = {"12345": "ewfnweoif"}
    mocker.patch(
        "rope.api.sessions.sessions",
        session_store,
    )
    response = test_client.delete("/session")
    assert response.status_code == 200
    assert session_store.get("12345") is None
```